src/ConnectionManager.ts

```diff
@@ -293,15 +293,24 @@ export class ConnectionManager extends EventEmitter {
      * @param type 
      */
     async getConnectionTargets(userId: string, type: string, filters: Record<string, unknown> = {}): Promise<unknown[]> {
-        if (type === GitLabRepoConnection.CanonicalEventType) {
-            if (!this.config.gitlab) {
-                throw new ApiError('GitLab is not configured', ErrCode.DisabledFeature);
-            }
-            if (!this.config.checkPermission(userId, "gitlab", BridgePermissionLevel.manageConnections)) {
-                throw new ApiError('User is not permitted to provision connections for GitLab', ErrCode.ForbiddenUser);
-            }
-            return await GitLabRepoConnection.getConnectionTargets(userId, this.tokenStore, this.config.gitlab, filters);
+        switch (type) {
+        case GitLabRepoConnection.CanonicalEventType:
+            this.validateConnectionTarget(userId, this.config.gitlab, "GitLab", "gitlab");
+            return await GitLabRepoConnection.getConnectionTargets(userId, this.tokenStore, this.config.gitlab!, filters);
```

Comment:
No non-nulls, prefer to throw if it's not defined. (As we did before)

Reply:
Ah, I see that's what the above is doing. I'd still return the config or something from the validateConnectionTarget to avoid non-nulls, because it basically ignores the typing system which will bite us one day.

Reply:
Thanks for the suggestion.  This is addressed in the latest commit.